Nos/Views/Discover/DiscoverContentsView.swift

```diff
@@ -3,7 +3,7 @@ import SwiftUI
 import Dependencies
 
 struct DiscoverContentsView: View {
-    @ObservedObject var searchController: SearchController
+    @State private var searchController: SearchController
```

Comment:
I think `@State` is unnecessary here. My understanding is that SwiftUI essentially wraps every call to `body` in `withObservationTracking` which will detect reads on any `@Observable` object in the body, and then trigger a new call to `body` when any of the read properties change. `@State` is only necessary now if you need the object to match the lifetime of the View. In this case the `searchController`'s lifecycle is managed by its superview. I tried removing it and everything seemed to work for me. 

```suggestion
    private var searchController: SearchController
```